citar.el

```diff
@@ -706,7 +709,31 @@ If FORCE-REBUILD-CACHE is t, force reload the cache."
 
 (defun citar--extract-keys (keys-entries)
   "Extract list of keys from KEYS-ENTRIES alist."
-  (seq-map #'car keys-entries))
+  (seq-map (lambda (key-entry)
+             (if (consp key-entry) (car key-entry) key-entry))
+           keys-entries))
+
+(defun citar--ensure-entries (keys-entries)
+  "Return copy of KEYS-ENTRIES with every element a (KEY . ENTRY) pair.
+Each element of KEYS-ENTRIES should be either a KEY or a (KEY
+. ENTRY) pair.  If it is the former, look up the corresponding
+ENTRY and transform the element to a (KEY . ENTRY) pair."
+  (let ((candidates 'uninitialized))
+    ;; Get candidates only if some key has a missing entry, to avoid nasty
```

Comment:
Under what condition might this happen?

Reply:
If a function is being called interactively, then `keys-entries` will come from `citar-select-ref` and so all the entries will be present; there's no need to get candidates. Right now this is just a performance thing, because there's no situation in which `citar--ensure-entries` would be called during candidate generation. But I figured it's good to code defensively anyway.